js_modules/dagit/packages/core/src/runs/RunTimeline.tsx

```diff
@@ -21,61 +22,115 @@ const LABEL_WIDTH = 232;
 
 const ONE_HOUR_MSEC = 60 * 60 * 1000;
 
-type RunForJob = {
+export type RunForJob = {
   id: string;
-  status: RunStatus;
+  status: RunStatus | 'SCHEDULED';
   startTime: number;
   endTime: number;
 };
 
-interface Props {
-  jobs: {
-    [jobKey: string]: {
-      path: string;
-      runs: RunForJob[];
-    };
+export type JobMap = {
+  [jobKey: string]: {
+    label: string;
+    path: string;
+    runs: RunForJob[];
   };
+};
+
+interface Props {
+  jobs: JobMap;
   range: [number, number];
 }
 
 export const RunTimeline = (props: Props) => {
   const {jobs, range} = props;
+  const [width, setWidth] = React.useState<number | null>(null);
+  const observer = React.useRef<ResizeObserver | null>(null);
+
+  const containerRef = React.useCallback((node) => {
```

Comment:
I'd almost forgotten you could pass a function to `ref=`, this seems like a nice solution.